tools/analyze_failures.py

```
@@ -0,0 +1,74 @@
+    parser.add_argument("--manifest", type=Path, default=Path("dataset_manifest.csv"))
+    parser.add_argument("--out-dir", type=Path, default=Path("benchmark-results"))
+    parser.add_argument("--high-confidence", type=int, default=90)
+    parser.add_argument("--low-confidence", type=int, default=70)
+    args = parser.parse_args()
+    if not args.predictions.is_file() or not args.manifest.is_file():
+        raise SystemExit("Both predictions.csv and dataset_manifest.csv are required.")
+    report = analyze(read_csv(args.predictions), read_csv(args.manifest), args.high_confidence, args.low_confidence)
+    args.out_dir.mkdir(parents=True, exist_ok=True)
+    with (args.out_dir / "failure_analysis.csv").open("w", newline="", encoding="utf-8") as stream:
+        writer = csv.DictWriter(stream, fieldnames=FIELDS)
+        writer.writeheader(); writer.writerows(report["records"])
```

Ruff (0.16.1)

[error] Multiple statements on one line (semicolon)

(E702)